### PR TITLE
Amendments on PR1

### PR DESCRIPTION
**Pull request by rvanheest**
Some additions to PR1:
- `collectResults` can be defined for any `Iterable` rather than for just one implementation (`List`). Make it as general as possible here, otherwise some day we will need this method for a `ListBuffer`, a `NodeSeq`, a `Map`, a `Set`, or whatever! I also tested that this still worked with the amendment, btw.
- I really like the documentation here, especially with the examples! Great idea!
- Some fields in the documentation were not yet filled in, some I added those.
- I think we should be really precise in this library here, especially with our types. Therefore I added the `Unit` output types for the two logging methods.

Hope these give even greater value to the whole lib.

@janvanmansum are you going to replace the old code in the various projects with this code as well?

### CODE REVIEW

File: src/main/scala/nl/knaw/dans/lib/error/package.scala [hunk 1]
```diff
@@ -3,21 +3,23 @@ package nl.knaw.dans.lib
...
+   * @param throwables a collection of `Throwable`s
    */
-  class CompositeException(throwables: List[Throwable])
+  class CompositeException(throwables: Iterable[Throwable])
     extends RuntimeException(throwables.foldLeft("")(
       (msg, t) => s"$msg\n${getMessage(t)} ${getStackTrace(t)}"
     ))
 
-  implicit class ListTryExtensions[T](xs: List[Try[T]]) {
+  implicit class IterableTryExtensions[T](xs: Iterable[Try[T]]) {
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/lib/error/package.scala]: Makes sense. Come to think of it, we also need unit tests ...

Comment (rvanheest) [reply src/main/scala/nl/knaw/dans/lib/error/package.scala]: Completely agree! Every bit of this lib should be tested before bringing it in the development cycle!

**Comment by rvanheest** [comment]
wait, something is not right yet... If I do `Set(try1,try2,try3).collectResults` I would expect to get a `Try[Set[T]]`, but now I would get a `Try[Iterable[T]]` instead.

gonna fix this!

**Comment by rvanheest** [comment]
haven't found a complete solution. gonna work in this some more at home and tomorrow.

**Comment by rvanheest** [comment]
guess what, I did it! I do not exactly understand what it all means, but "_follow the types_" turns out to be pretty useful; even here!

**Comment by rvanheest** [comment]
Forgot to commit some things yesterday:
- I changed `Iterable` to `Traversable`, as the latter is a supertype of the former
- I wrote some tests that ensure:
  - if a `List[Try[T]]` only contains `Success` instances, it returns a `Success`
  - if a `List[Try[T]]` contains one or more `Failure`, it returns a `Failure` with the collected error report
  - the output of calling `collectResults` on a `M[Try[T]]` (where `M[X] <: Traversable[X]`) should be a `Try[M[T]]`, e.g. the input collection type should be the same as the output collection type
  - if a `List[Try[T]]` contains one or more `Failure`s and one or more of them are `CompositeException`s, the error messages from the latter ones are flattened and together with the other `Failure`s combined into a single `CompositeException`
- Based on the last test I changed the implementation of `collectResults`; see the [change log](https://github.com/DANS-KNAW/dans-scala-lib/pull/2/commits/c25e799a62a0e9d4b24083c8623ffd54f6dd207b#diff-bc7d96adb19a6c6f7072e2d299516c97R54)

**Pull request merged**